### PR TITLE
fix 1120 incorrect file path

### LINKED ISSUE #1120: Pylint gets file path wrong
Pylint seems to mangle file paths when its run from a directory in '/' which has the same name as a dir lower down in the hierarchy which contains a python file with linting errors.
### Steps to reproduce

A (probably not completely minimised) directory structure (__init__s are all empty):

```
deterer@deterer /grom % ls -R
.:
test

./test:
grom  __init__.py

./test/grom:
__init__.py  test

./test/grom/test:
foo.py  __init__.py


deterer@deterer /grom % cat test/grom/test/foo.py 
a = object()
a.bar()

```

Run pylint:

```
deterer@deterer /grom %  pylint --reports no --output-format parseable --msg-template=ERROR:{path}:{line}:{symbol}:{msg} test/grom/test/foo.py 
```
### Current behavior

Output has the wrong path.

```
No config file found, using default configuration
/usr/lib/python2.7/dist-packages/pylint/reporters/text.py:79: UserWarning: parseable output format is deprecated. This is equivalent to --msg-template={path}:{line}: [{msg_id}({symbol}), {obj}] {msg}
  'to --msg-template=%s' % (self.name, self.line_format))
************* Module test.grom.test.foo
ERROR:testtest/foo.py:1:blacklisted-name:Black listed name "foo"
ERROR:testtest/foo.py:1:missing-docstring:Missing module docstring
ERROR:testtest/foo.py:1:invalid-name:Invalid constant name "a"
ERROR:testtest/foo.py:2:no-member:Instance of 'object' has no 'bar' member
```
### Expected behavior

Output has the right path

```
No config file found, using default configuration
/usr/lib/python2.7/dist-packages/pylint/reporters/text.py:79: UserWarning: parseable output format is deprecated. This is equivalent to --msg-template={path}:{line}: [{msg_id}({symbol}), {obj}] {msg}
  'to --msg-template=%s' % (self.name, self.line_format))
************* Module test.grom.test.foo
ERROR:test/grom/test/foo.py:1:blacklisted-name:Black listed name "foo"
ERROR:test/grom/test/foo.py:1:missing-docstring:Missing module docstring
ERROR:test/grom/test/foo.py:1:invalid-name:Invalid constant name "a"
ERROR:test/grom/test/foo.py:2:no-member:Instance of 'object' has no 'bar' member
```
### pylint --version output

```
deterer@deterer /grom % pylint --version
No config file found, using default configuration
pylint 1.1.0, 
astroid 1.0.1, common 0.61.0
Python 2.7.6 (default, Jun 22 2015, 17:58:13) 
[GCC 4.8.2]
```
### Additional

I have also run this in Docker to try it with Ubuntu 16.04 and pylint 1.6.4

```
deterer@deterer /grom % docker build -t dock-pylint .                                                                                         
Sending build context to Docker daemon 6.144 kB
Step 1 : FROM ubuntu:16.04
 ---> 45bc58500fa3
Step 2 : RUN apt-get -qq update
 ---> Using cache
 ---> db0460086527
Step 3 : RUN apt-get -qq install python python-pip vim tree
 ---> Using cache
 ---> 88bac80f6c74
Step 4 : RUN pip install pylint
 ---> Using cache
 ---> acad50c8ef05
Step 5 : COPY . /grom
 ---> 2e7a96ea7765
Removing intermediate container d383244f20e2
Successfully built 2e7a96ea7765
deterer@deterer /grom % docker run  -w /grom dock-pylint pylint --reports no --output-format parseable --msg-template=ERROR:{path}:{line}:{symbol}:{msg} *
No config file found, using default configuration
************* Module Dockerfile
ERROR:Dockerfile:1:syntax-error:invalid syntax
************* Module test.grom.test.foo
ERROR:testtest/foo.py:3:trailing-newlines:Trailing newlines
ERROR:testtest/foo.py:1:blacklisted-name:Black listed name "foo"
ERROR:testtest/foo.py:1:missing-docstring:Missing module docstring
ERROR:testtest/foo.py:1:invalid-name:Invalid constant name "a"
ERROR:testtest/foo.py:2:no-member:Instance of 'object' has no 'bar' member
deterer@deterer /grom % docker run -w /grom dock-pylint pylint --version                                                                                 
No config file found, using default configuration
pylint 1.6.4, 
astroid 1.4.8
Python 2.7.12 (default, Jul  1 2016, 15:12:24) 
[GCC 5.4.0 20160609]
```

### PR DESCRIPTION
**Pull request by reverbc**
### Fixes / new features
- Fixes #1120 

Only replace `path_strip_prefix` for the first encountered instance.

### CODE REVIEW

**Comment by PCManticore** [comment]
Hey @reverbc Thanks for your patch! Would it be possible to also submit a test? We have similar tests that check for the output in `test_self.py`, you could take some inspiration from one of those.

**Comment by reverbc** [comment]
Hi @PCManticore sorry I forgot to add the test 🤦‍♂️  

While I add the test, I discover that in some tests the pytest fixture wasn't correctly invoked - the marker should be `@pytest.mark.usefixtures` instead of `@pytest.mark.usefixture`. I've fixed this in the latest commit.

Also I'll appreciate if you can help to review the test itself. Since this issue only occurs when the folder tree of test target has multiple `reporter.path_strip_prefix` (like `/SRC/some/path/SRC/test_target.py`), I have to dynamically generate folder tree and test target file. This might not look pretty but it does the job. I'm open to more elegant solution ;)

I also disable this test for Windows as it will always have drive letter in the beginning of `reporter.path_strip_prefix` so no folder tree like this can be created.

Thanks!

File: pylint/test/test_self.py [hunk 1]
```diff
@@ -458,3 +458,23 @@ def test_getdefaultencoding_crashes_with_lc_ctype_utf8(self):
         module = join(HERE, 'regrtest_data', 'application_crash.py')
         with _configure_lc_ctype('UTF-8'):
             self._test_output([module, '-E'], expected_output=expected_output)
+
+    @pytest.mark.skipif(sys.platform == 'win32', reason='only occurs on *nix')
+    def test_parseable_file_path(self):
```
Comment (PCManticore) [inline pylint/test/test_self.py]: We could probably already create this directory in `pylint/test/regrtest_data`. Then in the test, we only have to call `self._test_output`.

Comment (reverbc) [reply pylint/test/test_self.py]: @PCManticore thanks for the input, however I think pre-create directory in `pylint/test/regtest_data` is not able to reproduce this issue, as `reporter.path_strip_prefix` is the absolute path all the way from `/` to current folder.

Take Travis CI with Python 3.6 env as example, the `reporter.path_strip_prefix` is actually `/home/travis/build/PyCQA/pylint/.tox/py36/lib/python3.6/site-packages/pylint/test/regrtest_data/package/`. This path can be changed with different python version and different run time environment, that's why I decided to dynamically create it based on current environment.

Comment (PCManticore) [reply pylint/test/test_self.py]: Agreed, sounds like this is the best way to test this scenario.

**Pull request merged**